### ISSUE TITLE
(NSM) Implementation for MLP and related interfaces and specializations

### ISSUE THREAD

**Issue by venkataram-nv** [shader-slang/slang#7866]

**Comment by jkwak-work** [comment]
Venki is tring out and replicate examples from slangpy.

**Comment by venkataram-nv** [comment]
Implementation and testing currently being done in a separate repo: https://github.com/venkataram-nv/slangpy-neural

Status for implementation:
- [x] InlineVector
- [x] CoopVector
- [x] ReLU
- [x] LeakyReLU
- [ ] Sigmoid
- [ ] Tanh
- [x] FrequencyEncoder
- [ ] SphericalHarmonicsEncoder
- [ ] HashTableEncoder
- [x] MeanSquaredError
- [ ] MeanAbsoluteError
- [x] Adam
- [ ] AdamW
- [ ] SGD
- [ ] Momentum
- [x] FeedForward layer

Status for testing:
- [x] Basic InlineVector operations (agreement with PyTorch)
- [x] Basic InlineVector derivtives (agreement with PTorch)
- [x] ReLU
- [x] ReLU derivative (agreement with PyTorch)
- [x] LeakyReLU
- [x] LeakyReLU derivative (agreement with PyTorch)
- [x] FeedForward
- [x] FeedForward gradients (agreement with PyTorch)
- [x] FrequencyEncoder
- [x] FrequencyEncoder derivative (agreement with PyTorch)
- [x] Sequence of FeedForward (i.e. a small neural network)
- [x] Neural network gradients (agreement with PyTorch)
- [x] MeanSquaredError
- [x] MeanSquaredError derivative (agreement with PyTorch)
- [x] Adam (agreement with PyTorch)

Status for examples: Not started

**Comment by kaizhangNV** [comment]
Close this task, as we will need separate task to easily track each component.
